public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.ts

```diff
@@ -58,11 +58,25 @@ export function transformSceneToSaveModel(scene: DashboardScene, isSnapshot = fa
   if (body instanceof SceneGridLayout) {
     for (const child of body.state.children) {
       if (child instanceof DashboardGridItem) {
-        // handle panel repeater scenatio
-        if (child.state.variableName) {
-          panels = panels.concat(panelRepeaterToPanels(child, isSnapshot));
+        let child_ = child;
+        // If we're saving while the panel editor is open, we need to persist those changes in the panel model
+        if (
+          child.state.body instanceof VizPanel &&
+          state.editPanel?.state.vizManager &&
+          state.editPanel.state.vizManager.state.sourcePanel.resolve().state.key === child.state.body.state.key
```

Comment:
is there a reason for comparing key here? would not state.editPanel?.state.vizManager.state.sourcePanel.resolve() === child always be enough?